Khan/perseus
# Throttle events in polygon

## Summary:

Throttles the events in polygon to increase performance and decrease CPU load. This is especially important for Safari. This change lowers CPU usage from 87% to 30% on a M3 macbook. The increase in performance will be more on phones and iPads where this will also decrease energy usage.

40 FPS was chosen as a mark that feels response and good to use but doesn't put too much stress on the machine.

Issue: https://khanacademy.atlassian.net/browse/LEMS-2185

## Test plan:

To test before:
- Load the dev environment on Safari
- Open the Timelines pane and start recording
- The CPU usage should be high (it'll vary based on your machine)

To test after:
- Repeat test above with this change
- **The CPU usage should be lower**
